### PR TITLE
Automate openapi server generation

### LINKED ISSUE #309: Automate OpenAPI Server Generation
**Is your feature request related to a problem? Please describe.**
Right now the OpenAPI server generation is not automatically triggered therefore we could miss some changes while updating the openapi spec.

**Describe the solution you'd like**
Include the openapi server re-generation into the `gen` target.
Make into conditional, such that it is really re-generated if and only if there are changes in the openapi spec.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by lampajr**
Fixes https://github.com/opendatahub-io/model-registry/issues/309

In addition it fixes the `type_assert.patch` which was _corrupted_ by the pre-commit hook

## Description

Fixed the patch by restoring required blank lines and spaces.

Auto generate `AssertDocArtifact*` methods by running `make gen/openapi-server`.

## How Has This Been Tested?

Running `make clean gen gen/openapi-server` you shouldn't see any error and no uncomitted changes

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by tarilabs — DISMISSED** [review]
nice catch! thank you so much 🙏 is there a way we can ensure not to miss this in the future in your opinion in an easy way? 🤔 

atm I can only think of just programmatically do it, but I wonder if it's too much work... maybe we should put some comments in the ModelArtifact, DocArtifact that due to the way the codegen works, this method is to be expected?

**Comment by lampajr** [comment]
> is there a way we can ensure not to miss this in the future in your opinion in an easy way?

Well for the type_assert.patch issue I would remove it from the pre-commit formatting if possible, @isinyaaa is it possible to do that?

Regarding the `Assert*` methods, actually they are not used in the server (at least for some of them that I checked).. therefore we could also evaluate if we need that file in the future but yeah I think the best practice would be to always run `make gen/openapi-server` whenever we change the openapi spec (right now this is not automatically triggered but we could think to include it as well as we do for the client/model part)

**Comment by tarilabs** [comment]
> right now this is not automatically triggered

yes, this part.

**Comment by tarilabs** [comment]
> I would remove it from the pre-commit formatting if possible

try with adding ` exclude: '^patches/'` just at the very root of the `.pre-commit-config.yaml`.

(edit: I was not able to configure per-hook)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ repos:
       - id: detect-private-key
       - id: end-of-file-fixer
       - id: trailing-whitespace
+    exclude: '^patches/'
```
Comment (tarilabs) [inline .pre-commit-config.yaml]: did this work for you? I got a "wrong format" error message of sort? 🤔

Comment (lampajr) [reply .pre-commit-config.yaml]: Sorry I did not try that yet, I wrongly included that in the commit.. trying right now

Comment (lampajr) [reply .pre-commit-config.yaml]: As you suggested I tried adding that line at root of the file and it seems working.

Comment (tarilabs) [reply .pre-commit-config.yaml]: > As you suggested I tried adding that line at root of the file and it seems working.

🥳 🚀

Comment (tarilabs) [reply .pre-commit-config.yaml]: we can use analogously if the linter is going _beyond_ the Go ootb Linter hence colliding with the .patch, in case

Comment (lampajr) [reply .pre-commit-config.yaml]: +1 I agree

**Comment by lampajr** [comment]
Ok I was able to automate the openapi-server generation but there are some issues in the `build` workflow, somehow the build generates some uncomitted changes, instead I tried running the same command multiple times locally and I got the same result.

So for now I think we can just remember to run `make gen/openapi-server` whenever we made changes to the api spec

**Review by antisaling — DISMISSED** [review]
Thanks a lot for fixing that!

**Comment by lampajr** [comment]
As suggedted by @tarilabs I was able to automate the openapi server generation only if there are some changes (even if staged) in the `api/openapi/model-registry.yaml` file.

This way the server generation won't happen during the GHA.

**Comment by lampajr** [comment]
Additional comment is that you can always force the re-generation by setting `FORCE_SERVER_GENERATION` env variable, e.g., by running `make FORCE_SERVER_GENERATION=true gen/openapi-server`

**Review by tarilabs — APPROVED** [review]
Awesome, thank you so much, and glad some of the suggestions were relevant! 🚀 👍

**Pull request merged**